File: Data Structures & Algorithms/validate-parentheses/submission-4.py
class Solution:
    def isValid(self, s: str) -> bool:
        braces = {"(":")","[":"]","{":"}"}
        stack = []

        for char in s:
            if char in braces:
                stack.append(char)
                continue
            if stack and char not in braces.keys() and braces[stack[-1]] == char:
                stack.pop()
            else:
                return False
        return True if not stack else False
            

File: Data Structures & Algorithms/minimum-stack/submission-4.java
class MinStack {
    private Stack<Integer> stack;
    private Stack<Integer> minStack;

    public MinStack() {

        stack = new Stack<>();
        minStack = new Stack<>();
    }
    
    public void push(int val) {
        stack.push(val);
        if (minStack.size() ==0){
            minStack.push(val);
        }
        else{
            int t = Math.min(val,minStack.peek());
            minStack.push(t);
        }
        
    }
    
    public void pop() {
        stack.pop();
        minStack.pop();
        
    }
    
    public int top() {
        return stack.peek();
        
    }
    
    public int getMin() {
        return minStack.peek();
        
    }
}
